bin/install

```diff
@@ -18,23 +19,109 @@ error_exit() {
   exit "${2:-1}"
 }
 
-platform() {
+get_platform() {
   local -r os=$(
     case "$(uname | tr '[:upper:]' '[:lower:]')" in
       darwin) echo "osx" ;;
       *) echo "linux" ;;
     esac
   )
 
-  echo "${os}-amd64"
+  echo "${os}"
+  return
 }
+get_arch(){
```

Comment:
The `get_arch()` function returns the architecture of the system, returning `'32'` for x86 and compatibles, as well as `'64'` for x86_64

`error_exit`s if the release if `uname -m` does not return one of `x86_64`, `i386`, or `i686`